components/clarity-repl/src/repl/session.rs

```diff
@@ -657,14 +662,60 @@ impl Session {
         Ok((execution, contract_identifier))
     }
 
+    fn run_wasm_and_clarity(
+        &mut self,
+        snippet: String,
+        eval_hooks: Option<Vec<&mut dyn EvalHook>>,
+        cost_track: bool,
+    ) -> Result<ExecutionResult, Vec<Diagnostic>> {
+        // @todo: we should take another approach than a regex to parse it
```

Comment:
The parser could be useful here.

Reply:
Instead though, it is probably better for the wasm runtime to provide an interface that would just allow you to pass it the snippet and run that.